docs/redesign-proposal.md

```diff
@@ -0,0 +1,93 @@
+# Navigator design
+
+Navigator as a system intends to provide a way to easily provision, monitor and
+manage complex systems on top of cluster schedulers, initially targetting
+Kubernetes and OpenShift.
+
+## Architecture
+
+All communication between components of the Navigator system should communicate
+through the `navigator-apiserver`. This allows for a consistent audit trail,
+recorded in a central place. Monitoring services such as `kube-state-metrics`
+can then be used to track changes to the contents of the API server to be
+consumed by something like Prometheus.
```

Comment:
Having read further, it seems that one key reason for `navigator-api-server` is that OpenShift (and other OFs) don't support custom resource definitions. In that case, maybe state it here.